bundle/config/mutator/resolve_variable_references.go

```diff
@@ -56,29 +62,56 @@ func (m *resolveVariableReferences) Apply(ctx context.Context, b *bundle.Bundle)
 			// Future opportunity: if we lookup this path in both the given root
 			// and the synthesized root, we know if it was explicitly set or implied to be empty.
 			// Then we can emit a warning if it was not explicitly set.
-			return dyn.GetByPath(normalized, path)
-		}
+			v, err := dyn.GetByPath(normalized, path)
+			if err != nil {
+				return dyn.InvalidValue, err
+			}
 
-		// Resolve variable references in all values.
-		root, err := dynvar.Resolve(root, func(path dyn.Path) (dyn.Value, error) {
-			// Rewrite the shorthand path ${var.foo} into ${variables.foo.value}.
-			if path.HasPrefix(varPath) && len(path) == 2 {
-				path = dyn.NewPath(
-					dyn.Key("variables"),
-					path[1],
-					dyn.Key("value"),
-				)
+			// Check if the variable we're looking up has a lookup field in it
+			lookupV, err := dyn.GetByPath(normalized, path[:len(path)-1].Append(dyn.Key("lookup")))
```

Comment:
This is ambiguous. It also errors if we have:

```yaml
obj:
  bar: xyz
  lookup: ""
```

And then interpolate `${obj.bar}`.